packages/sdk-rtl/src/lookerSDKError.ts

```diff
@@ -0,0 +1,185 @@
+  AdditionalErrorOptions
+> = ErrorParameters extends [(infer Message)?]
+  ? [Message?, AdditionalErrorOptions?]
+  : ErrorParameters extends [(infer Message)?, (infer ErrorOptions)?]
+  ? [Message?, (ErrorOptions & AdditionalErrorOptions)?]
+  : ErrorParameters extends [
+      (infer Message)?,
+      (infer ErrorOptions)?,
+      ...infer Rest
+    ]
+  ? [Message?, (ErrorOptions & AdditionalErrorOptions)?, ...Rest]
+  : never
+
+interface IErrorDetail {
+  field?: string | null
+  code?: string | null
+  message?: string | null
+  documentation_url: string | null
+}
+
+// This specifies SDK custom error options
+interface ILookerSDKErrorOptions {
+  errors?: IErrorDetail[]
+  documentation_url?: string | null
+}
+
+interface ILookerSDKErrorConstructor {
+  new (
+    ...args: AugmentErrorOptions<
+      ConstructorParameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+  (
+    ...args: AugmentErrorOptions<
+      Parameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+}
+
+// The subclass and function expression's name should match, so that stack traces look clean.
+// We bind it to a local identifier for clarity, and to perform a type assertion.
+export interface LookerSDKError extends Error {
+  errors?: IErrorDetail[]
+  documentation_url?: string | null
+}
+
+export const LookerSDKError: ILookerSDKErrorConstructor =
+  /* #__PURE__*/ (() => {
+    'use strict'
+    const LookerSDKErrorConstructor = function LookerSDKError(
+      this: LookerSDKError | undefined,
+      ...[
+        message,
+        { errors = [], documentation_url = '', ...errorOptions } = {},
+        ...rest
+      ]: AugmentErrorOptions<
+        ConstructorParameters<ErrorConstructor>,
+        ILookerSDKErrorOptions
+      > &
+        AugmentErrorOptions<
+          Parameters<ErrorConstructor>,
+          ILookerSDKErrorOptions
+        >
+    ) {
+      // The `super()` call. At present, Error() and new Error() are
+      // indistinguishable, but use whatever we were invoked with in case
+      // that ever changes.
+      const error = this
+        ? new Error(
+            message,
+            // we have to suppress a type error here if TypeScript
+            // doesn't know es2022's two-argument Error constructor
+            // @ts-ignore-error
+            errorOptions,
+            ...rest
+          )
+        : Error(
+            message,
+            // @ts-ignore-error
+            errorOptions,
+            ...rest
+          )
+
+      // Object.setPrototypeOf() is necessary when extending built-ins,
+      // since Error.call(this, message, errorOptions, ...rest) doesn't
+      // set up the prototype chain the way it would with a user-defined
+      // class.
+      Object.setPrototypeOf(
+        error,
+        this ? Object.getPrototypeOf(this) : LookerSDKError.prototype
+      )
+
+      // imitates the non-enumerability of the 'message' and 'stack'
```

Comment:
Have you thought about whether you want these to be enumerable? Is it better for your API to match the behavior of the non-enumerable `message` and `stack` fields, or not?

Should `Object.keys(someError)` return `['errors', 'documentation_url']`, or should it return `[]`?

Do you want `for (const field in someError) {}` to iterate your custom fields, or to be a no-op?

I don't have a position on this, but you should talk about it with the team before committing one way or the other.